File: 0090-subsets-ii/0090-subsets-ii.java
class Solution {
    List<List<Integer>> res = new ArrayList<>();
    List<Integer> path = new ArrayList<>();;
    public List<List<Integer>> subsetsWithDup(int[] nums) {
       Arrays.sort(nums);
       backtracking(nums, 0);
       return res;
    }

    private void backtracking(int[] nums, int start) {
       res.add(new ArrayList<Integer>(path));

       for (int i = start; i < nums.length; i++) {
           if (i > start && nums[i] == nums[i - 1]) {
               continue;
           }
           path.add(nums[i]);
           backtracking(nums, i + 1);
            path.remove(path.size() - 1);
       }
    }
}

File: 0090-subsets-ii/0090-subsets-ii.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var subsetsWithDup = function(nums) {
    const res = [];
    const subset = [];
    nums.sort((a, b) => a - b);
    
    function backtrack(start) {
        res.push([...subset]);
        
        for (let i = start; i < nums.length; i++) {
            if (i !== start && nums[i] === nums[i - 1]) {
                continue;
            }
            
            subset.push(nums[i]);
            backtrack(i + 1);
            subset.pop();
        }
    }
    
    backtrack(0);
    return res;
};